### PR TITLE
Proposal: Adding LiveView Component to Heroicons

### PR DESCRIPTION
**Pull request by bmalum**
First of all - thank you for providing the base package. I did almost the same, for my component library and then noticed your hex package. 

I decided to added my (tbh very small part) to your package. I can fully understand if you think that this should not be included in your package, because it adds more dependencies. 

Let me know if there is something to improve!

### CODE REVIEW

File: lib/heroicons/live_view_components.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  """
+  @moduledoc since: "0.3.0"
+
+  @callback render(assigns :: Phoenix.LiveView.Socket.assigns()) ::  Phoenix.LiveView.Rendered.t()
+  @doc false
+
+
+  def render(%{type: :outline, class: class, icon: icon} = assigns) do
+    ~L"""
+    <%=  apply(Elixir.Heroicons.Outline, icon, [%{class: class}]) |> Phoenix.HTML.raw %>
```
Comment (mveytsman) [inline lib/heroicons/live_view_components.ex]: I think the functions return html safe data, is the call to `raw` necessary?

**Review by mveytsman — COMMENTED** [review]
Thank you for this PR!

I've been thinking about doing this as well so I really appreciate it. 

I'm trying to figure out what the situation is where I want a component in a LiveView app for these. E.g. in my LiveView app I do

```
<%= Heroicons.Solid.server(class: "ml-6 w-12 h-12 text-green-400" %>
```

What does adding live components buy us? We're now being more verbose as

```
<%= live_component @socket, Heroicons.LiveComponent, type: :outline, icon: :server, class: "ml-6 w-12 h-12 text-green-400" %>
```

We could gain some more efficient diffing, e.g if you change the class in a diff a component would be able to send a smaller diff, but I think we'd need to get the SVG into the render function with the classes as part of the assigns for that to work.

@bmalum were you writing LiveView code where having a component would have been really helpful?

**Comment by bmalum** [comment]
I usually also do use `<%= Heroicons.Solid.server(class: "ml-6 w-12 h-12 text-green-400" %>` but now I have to use Heroicons in a very long dynamic List for showing a state (list is used by multiple users) - so I was thinking about that every item in this list can be a component and update it with `send_update` or I can go even smaller and just update the icon.

Maybe I'm wrong but I thought to get a minimal diff as possible, best way is to use components and `send_update` this was my use case. Maybe a niche 😉

File: lib/heroicons/live_view_components.ex (hunk 1 above)
Comment (bmalum) [reply lib/heroicons/live_view_components.ex]: good point - you are right - I updated this part.

**Comment by mveytsman** [comment]
That makes sense!

I have a feeling that as written the component diffs would be the same and would have the entire SVG in it even if you're updating just the class.

**Comment by mveytsman** [comment]
I think the stateless function components (https://github.com/phoenixframework/phoenix_live_view/pull/1428) are exactly what we need here. I'm going to close this in favor or https://github.com/mveytsman/heroicons_elixir/issues/3

**Pull request closed without merging**